sympy/functions/special/bsplines.py

```diff
@@ -202,3 +202,60 @@ def bspline_basis_set(d, knots, x):
     """
     n_splines = len(knots) - d - 1
     return [bspline_basis(d, knots, i, x) for i in range(n_splines)]
+
+def interpolating_spline(d, x, xv='x', yv='y'):
+    """Return B-splines at ``x`` of degree ``d`` for given ``xv`` and corresponding ``yv`` values.
+
+    This function returns a list of Piecewise polynomials that are the
+    B-splines of degree ``d`` for the given points. This function
+    calls ``bspline_basis_set(n, knots, x)`` for different values of ``n``.
+
+    Examples
+    ========
+
+    >>> from sympy interpolating_spline
+    >>> from sympy.abc import x
+    >>> d = 2
+    >>> splines=interpolating_spline(d, x, [1,2,3], [1,2,3])
+    >>> splines
+    [4*Piecewise((-x**2/4 + x - 3/4, (x >= 1) & (x <= 3)), (0, True)) 
```

Comment:
This example shows this function is not ready for release. The spline in this example should actually be simply x. The answer is unnecessarily complicated because the Piecewise objects are not combined. This was among the to-do points I listed when I raised the issue, and it is not done yet.